AUTHORSHIP.md

```diff
@@ -0,0 +1,71 @@
+# Authorship guidelines for academic papers and software archives
+
+First of all, we are deeply thankful to everyone who has helped make Fatiando a
+Terra what it is today. Our goal for this document is to establish guidelines
+for giving credit to contributors for their work.
+To do so, we will attempt to define:
+
+- Fair and diverse ways of providing recognition for contributors' efforts.
+- Define _contributions_ in a broad way: writing code and/or documentation,
+  providing ideas, fostering the community, etc.
+
+The following are the ways in which individuals who have contributed will be
+recognized.
+
+> **Note**: These policies are not set in stone and may be changed to
+> accommodate the growth of the project or the preferences of the community.
+
+## The `AUTHORS.md` file
```

Comment:
I 100% agree that in order for contributors to be included they need to add themselves to `AUTHORS.md`. If I remember correctly, this is on the checklist for pull requests?

Reply:
Thanks for the feedback @jessepisel!
It's not exactly on the pull requests checklist, but your memory is on a good track: after the first PR of a new contributor is merged, a bot instruct them to open a new PR to add themselves to `AUTHORS.md`.

I've seen that most people ignore or forget to do it. So this is one of the things we want to change: we will ask them to add their full names (with optional affiliation and ORCID) to `AUTHORS.md` from within the PR before it's merged. This would help the new contributors because they won't need to open a second PR (what could be challenge if you are new on GitHub) and help to remind the maintainers to explicitly ask the new contributors to add themselves.

Reply:
Sounds like a useful addition for the checklist in the PR template. Might be useful to create an org-wide default for this (and possibly other community files), see:

https://help.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file

Reply:
Thanks @hugovk! In fact we were thinking of adding the PR template here (to [fatiando/contributing](https://github.com/fatiando/contributing)) where all the community files that should be shared by every Fatiando package live.

> Might be useful to create an org-wide default for this (and possibly other community files), see:
> https://help.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file

Thanks for the info. I wasn't aware of the default community files repository. It seems a very nice option. Although I'm curious if community files are somehow automatically updated if any change is made on the `.github` repository.

FYI, the [fatiando/contributing](https://github.com/fatiando/contributing) repo satisfy the same needs as the `.github` repo. While any new Fatiando repository must be created through the [fatiando/package-template](https://github.com/fatiando/package-template), a cookiecutter template.
The nice thing about [fatiando/contributing](https://github.com/fatiando/contributing) is that whenever we push any new commit to its `master`, @fatiando-bot automatically opens a PR on every Fatiando library asking to incorporate the changes on the community files.

Maybe we could think of renaming this repo to `.github` and keep @fatiando-bot doing its job.

By the way,  @leouieda, I'm curious about how the community files are generated when creating a new repository from [fatiando/package-template](https://github.com/fatiando/package-template). Are they hardcoded or are they fetched from [fatiando/contributing](https://github.com/fatiando/contributing)?

Reply:
That sounds quite nifty that @fatiando-bot will do that syncing!

Yes, you wouldn't need to do any syncing when things are in `.github`, it should just read it straight from there (unless a repo has its own version of the file).

Reply:
I've just experienced a little with the `.github` repository.
So the community files are just linked on any library that belongs to the project. For example, when opening a new Issue, a link to the Contributor Guidelines appear. But the community files are not automatically included on them.
I think we may want to avoid that and explicitly include the community files on each repo.
What do you think?

Reply:
I've only used it for `ISSUE_TEMPLATE.md` and `PULL_REQUEST_TEMPLATE.md`, where it pre-fills the new issue or PR edit box. For example, in a repo which doesn't have its own `ISSUE_TEMPLATE.md`, it pre-fills this one from the `.github` repo:

![image](https://user-images.githubusercontent.com/1324225/75043539-ecb99880-54c8-11ea-97b6-0b59d13b2906.png)

For the guidelines and code of conduct, if I click to create a new issue in a repo here, I see:

![image](https://user-images.githubusercontent.com/1324225/75043919-710c1b80-54c9-11ea-9633-592d9092b293.png)

They link to:
* https://github.com/fatiando/harmonica/blob/master/CONTRIBUTING.md
* https://github.com/fatiando/harmonica/blob/master/CODE_OF_CONDUCT.md

Is that acceptable if they linked here?
* https://github.com/fatiando/.github/blob/master/CONTRIBUTING.md
* https://github.com/fatiando/.github/blob/master/CODE_OF_CONDUCT.md

Reply:
I see!
My thoughts are:
- All community files (`CODE_OF_CONDUCT.md`, `CONTRIBUTING.md`, `MAINTENANCE.md`, etc) should live inside each repository. I'm thinking it's better this way because releases uploaded to archives will have the full documentation of the repository. Besides, I think JOSS requires to have this on the repository, am I right?
- Maybe we could store PR and ISSUES templates on a new `fatiando/.github` repo. So we don't have to bother to update the ones that live inside the repositories if we make changes. Besides, there's no point of archiving these files, they are useful within GitHub.

Sorry for changing my mind on every comment, I'm just sharing thoughts and ideas to see if they are accurate or if there are better options.

Reply:
I don't know if it's a JOSS requirement, but yes, that sounds like a good approach!

Reply:
@hugovk I didn't know about the `.github` repo either. Nice!

I agree with @santisoler about leaving the community files where they are (it is a JOSS requirement) and creating a new `.github` repo to host the templates. It would be fine if they link to the files in this repo (`contributing`) instead of the ones on the individual repos. 

Talking to @djhoese on twitter, he pointed out that they can get a lot of people to add themselves to their authors file without much prodding. I'd be in favour of adding it to the checklist instead of having it as a message from the bot (which nobody seems to read).